templates/README.md

```diff
@@ -28,53 +26,84 @@ The `config.yaml` file has two keys:
+
+```yaml
 templates:
-  - from: templates/generic-component.yaml
+  - from: templates/generic.tmpl
     render:
-      - to: ../prow/jobs/kyma/components/console-backend-service/console-backend-service-generic.yaml
-        values:
-          <<: *kyma_generic_component
-          path: components/console-backend-service
-          since: "1.6"
-      - to: ../prow/jobs/kyma/tests/console-backend-service/console-backend-service-tests-generic.yaml
-        values:
-          <<: *kyma_generic_component
-          path: tests/console-backend-service
-          since: "1.6"
-
+      - to: ../prow/jobs/kyma/components/application-gateway/application-gateway-generic.yaml
+        jobConfigs:
+          - repoName: "github.com/kyma-project/kyma"
+            jobs:
+              - jobConfig:
+                  path: components/application-gateway
+                  args:
+                    - "/home/prow/go/src/github.com/kyma-project/kyma/components/application-gateway"
+                  run_if_changed: "^components/application-gateway/|^common/makefiles/"
+                  release_since: "1.7"
+                inheritedConfigs:
+                  global:
+                    - "jobConfig_default"
+                    - "image_buildpack-golang"
+                    - "jobConfig_generic_component"
+                    - "jobConfig_generic_component_kyma"
+                    - "extra_refs_test-infra"
+                  preConfigs:
+                    - "jobConfig_presubmit"
+                  postConfigs:
+                    - "jobConfig_postsubmit"
+                    - "disable_testgrid"
+        - to: ../prow/jobs/kyma/tests/application-gateway-tests/application-gateway-tests-generic.yaml
+        jobConfigs:
+          - repoName: "github.com/kyma-project/kyma"
+            jobs:
+              - jobConfig:
+                  path: tests/application-gateway-tests
+                  args:
+                    - "/home/prow/go/src/github.com/kyma-project/kyma/tests/application-gateway-tests"
+                  run_if_changed: "^tests/application-gateway-tests/|^common/makefiles/"
+                  release_since: "1.7"
+                inheritedConfigs:
+                  global:
+                    - "jobConfig_default"
+                    - "image_buildpack-golang"
+                    - "jobConfig_generic_component"
+                    - "jobConfig_generic_component_kyma"
+                    - "extra_refs_test-infra"
+                  preConfigs:
+                    - "jobConfig_presubmit"
+                  postConfigs:
+                    - "jobConfig_postsubmit"
+                    - "disable_testgrid"
 ```
 
 ### Component templates
 
+Component jobs are defined similarly to a regular job, with the exception that `name` field has to be empty, as it will be generated by the Render Templates tool; and `path` value has to be set.
```

Comment:
```suggestion
Component jobs are defined similarly to a regular job, with the exception that the `name` field must be empty (because the name is generated by the Render Templates tool), and the `path` value must be set.
```